File: bindings/swift/examples/TransactionsWithFunction.swift
// Copyright (c) 2026 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

import IotaSDK

@main
struct TransactionsWithFunctionExample {
  static func main() async throws {
    let client = GraphQlClient.newTestnet()
    let transactions = try await client.transactions(
      filter: TransactionsFilter().withFunction(function: "0x3::iota_system::request_add_stake"))
    for transaction in transactions.data {
      print("Digest:", transaction.transaction.digest().toBase58())
    }
  }
}

File: bindings/kotlin/examples/TransactionsWithFunction.kt
// Copyright (c) 2025 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

import iota_sdk.GraphQlClient
import iota_sdk.TransactionsFilter
import kotlinx.coroutines.runBlocking

fun main() = runBlocking {
    try {
        val client = GraphQlClient.newTestnet()
        val transactions =
            client.transactions(
                TransactionsFilter().withFunction("0x3::iota_system::request_add_stake")
            )
        for (transaction in transactions.data) {
            println("Digest: ${transaction.transaction.digest().toBase58()}")
        }
    } catch (e: Exception) {
        e.printStackTrace()
        kotlin.system.exitProcess(1)
    }
}
